rclcpp_components/src/component_container_mt.cpp

```diff
@@ -22,8 +22,22 @@ int main(int argc, char * argv[])
 {
   /// Component container with a multi-threaded executor.
   rclcpp::init(argc, argv);
+
+  auto options = rclcpp::NodeOptions{}.start_parameter_event_publisher(false);
   auto exec = std::make_shared<rclcpp::executors::MultiThreadedExecutor>();
-  auto node = std::make_shared<rclcpp_components::ComponentManager>(exec);
+  auto node = std::make_shared<rclcpp_components::ComponentManager>(
+    exec, "ComponentManager", options);
+  if (node->has_parameter("thread_num")) {
+    const auto thread_num = node->get_parameter("thread_num").as_int();
+    if (thread_num < std::thread::hardware_concurrency()) {
```

Comment:
why checking this?
I would use as many threads as the user indicated.

We could limit it to a maximum of `std::thread::hardware_concurrency()`, but we should print a warning when that happens.

Reply:
Default multi thread executor use `std::thread::hardware_concurrency()` as thread number, so it is redundant to create instance again if `thread_num` is greater than `std::thread::hardware_concurrency()`.